### DISCUSSION TITLE
Sprint 15

### DISCUSSION THREAD

**Discussion by Justus-at-Tazama** [tazama-lf/General-Issues#328] [General]
## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

### @cshezi - Cebolenkosi Shezi

**Yesterday:**

**Today:**

### @vorsterk - Kyle Vorster

**Yesterday:**

**Today:**

### @rtkay123 - Rodney Kanjala

**Yesterday:**

**Today:**

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

**Comment by Justus-at-Tazama** [comment]
## 2023/12/04

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Completed merging changes for the removal of the status on the channel object. Completed the routing of the flow based on the network map.

**Today:** Demo video and base fixtures ticket.

### @vorsterk - Kyle Vorster

**Yesterday:** Attempting to set up a dev environment with the persistent volumes. Super frustrating. Disks refuse to be created by script - had to force a manual creation. Still couldn't mount the disks in Azure. Gave up and created a new dev cluster. Renamed platform to EKUTA. Assigned appropriate names to cluster.

**Today:** Complete the installation and run a performance benchmark.

**Backlogged**: Updates of component versions.

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

### @rtkay123 - Rodney Kanjala

**Yesterday:** Code review session on logging changes. Code reviews. Assisted JP with ProtoBuf.

**Today:** Implement logging recommendations. Updating processors to add logging context. Demo prep.

### @johanfol - Johan Foley

**Yesterday:**

**Today:** Planning for the Linux Foundation (evaluate priority for code review). Planning for pacs.008 pivot.

## Parking lot

1. Optimizing the typology processor payload. A change was introduced to hold back the publishing of the `transaction` object from the typology processor result until the final typology is evaluated. @shezi to create a ticket and quarantine the change and separate the change from the update to the channel payload to remove the alert status. Change to be evaluated at the design authority against design principles (specifically statelessness).

**Reply by shezi** [reply]
wrong @shezi, I guess you meant @cshezi

**Comment by Justus-at-Tazama** [comment]
## 2023/12/05

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Quarantined the payload update for the typology processor result. Implemented the base-fixtures for tests. Should be updated as we go. Reviewed indexes on the pseudonyms collection.

**Today:** Start pushing base-fixtures tests and wrap up the ticket. This would be for the core processors - not the rules.

### @vorsterk - Kyle Vorster

**Yesterday:** Performance benchmarking. Great results. Achieved up to 2129 evaluations per second. Bottleneck on the TADProc as the bottleneck is pushed back from the typology processor. Getting a conflict on the `EndToEndId` for indexing on the `pseudonyms.transactionRelationship` edge collection. Added new Redis cluster to the server. Arango is now the new bottleneck as well due to processing overhead in the data preparation service.

**Today:** More benchmarking. Resolving the indexing. 

### @rtkay123 - Rodney Kanjala

**Yesterday:** Logs went down. Recreate the index on Elastic and recovered. Had to recreate the data views. Updated CRSP with context logging. Added Docker files and service deployment to work with the side-car.

**Today:** Completed remaining processors. Adding context on log events. Document Elastic indexes.

### @johanfol - Johan Foley

**Yesterday:** pacs.008 pivot design. Performance benchmarking. Cloned ArangoDB queries onto dev and ran then in parallel to the ingestion. Estimates on dev were far greater than estimates on SB2.

**Today:** Performance benchmarking. Compare and investigate selectivity estimates on dev.

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

October dip in performance due to pacs-002-relative rules.

**Comment by Justus-at-Tazama** [comment]
## 2023/12/05

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Worked on end-to-end testing of the NATS routing according to the network map. Reviewed PRs for the changes to the library for the event-sidecar.

**Today:** More PRs, wrap up NATS routing.

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on the local deployment of EKUTA. Reviewed Arango indexes. Tried to maximize resource utilization. TP wasn't working, needed to be fixed. Worked on upgrading Jenkins on the Sybrin side. Helm-charts on the fly! Assisted Rodney with the Lumberjack deployment. 

**Today:** Wrap up Jenkins. Max out CPUs on all processors. More performance benchmarking.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Debugging the typology processor via logging. Updated the pipeline scripts for the rules on SB1. Logging can reference the specific rule that is running instead of the rule executor. Functional logging service working in spite of issues with the GitHub token. Issues on the typology processor in progress.

**Today:** Wrap up TADProc logging and the typology processor and PRs and documentation for Kibana for Kyle.

### @johanfol - Johan Foley

**Yesterday:** RFP stuff.

**Today:** Deferred admin and planning.

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by vorsterk** [comment]
## 2023/12/06

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Continue with Nats routing using network map for TADP the unit test were giving problem proposed the split of tests and it seems like the headache came from addOneGetCount from redis call some the changes brought focus on the unit test not being self contained and The tested the branch on SB1, Couple from PRs for Rodney.

**Today:** Continuing with CRSP to just use the pre-fixing that is suggested by the ticket completing the ticket and documenting the solutions

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on Jenkins on Local Sybrin servers . When trying to setup up the rest of the jobs, the port forwarded instance of jenkins kept on timing out so it was frustrating because I would lose my work . I asked the internal guys to setup my DNS endpoints so I could work timeout free. They setup the endpoints but for some reason its not working. Looked at adding the max cpu usage of the processors configuration on jenkins jobs to replace the values in the dockerfiles but for some reason its not in the dockerfiles anymore. I need these values to consume more resources on our local nodes to get the maximum performance out.

**Today:** Create a task for Shezi to add the env variable to the dockerfiles again. Figure out why the endpoints arent working with Chris. Finalize the Jenkins jobs and Check arango indexes for performance increases.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Reviewed and merged PRs from Shezi. Fixed tests which needed some updating because of library changes. Revised and published documentation draft for the new logging framework. Rolled out new logging framework for the new processors

**Today:** Looking at some types. Log context (message id, in particular) was not attached to some logs in some processor because of loose typing on the processor's side. Use of unknown type should probably be replaced with FRMSMessage type in the library. Figure out why lumberjack token is not accepted over on Jenkins and sort that out. Finish docs for Kyle's indices.
Also just in, some rules (rule-executer) are not logging properly. Going to dive in there and see why

### @johanfol - Johan Foley

**Yesterday:**

**Today:** 

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by vorsterk** [comment]
## 2023/12/07

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Fully rolled out the Nats changes through the networkmap , chaos with everyone working on Sb1 , Completed the ticket and the documentation draft. 

**Today:** Complete Documentation today and then Justus can check. Let Justus know Sybrin config manager not working with new changes. Pickup new ticket for MaxCpu

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on getting the DNS external endpoints up on Sybrin locals , it was a certificate issue . Then worked on the Jenkins worker stuff to deploy locally but will need another sybrin blade to run Jenkins worker. WOrked on Jenkins job and Sortedour arango index issue - there shouldnt be a space in the fields that are listed.

**Today:**  Run some tests with the changes for maxcpu and the arango indexes.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Got a report that some rules weren't sending through to elastic , made batch size configurable - batch size research . CRSP wasn't attaching log context - failing on the test context.

**Today:** Go over documentation for indexes with Kyle . Linting/CI adding to lumberjack and sidecar.

### @johanfol - Johan Foley

**Yesterday:** 

**Today:** 

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by vorsterk** [comment]
## 2023/12/11

## Attendees
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Completed the dockerfile changes ticket - for Kyle . 

**Today:** Upskilling because no tasks in the to do , add more steps on the CI and github action NATS and Arango , API testing on Jenkins.

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on local servers making the jenkins pipelines and running some performance tests on the changes for the arango indexes and saw a performance increase.

**Today:**  Running more tests and looking at our application updates for better performance .

### @rtkay123 - Rodney Kanjala

**Yesterday:** Created and worked on code style tickets for the new processors. Implemented code styles for event-sidecar and updated workflows repo to separate lints and tests. Going to roll out changes to the rest of the processors today, starting with lumberjack

**Today:** Creating test suites for new processors (this is hard though lol not sure what to test for)

### @johanfol - Johan Foley

**Yesterday:** On holiday

**Today:** On holiday

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by cshezi** [comment]
## 2023/12/12

## Attendees
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Worked on Github workflow to do API testing had a bit of struggle with contanerized rest-nats-proxy's network and solved locally seems that same issue is coming for github runner, newman intergrated for db setup, and CRSP service startup is complete

**Today:** Debugging contanerized rest-nats-proxy's network on github runners and connecting it to CRSP for API testing with newman so we can log result to PR as comment

### @vorsterk - Kyle Vorster

**Yesterday:** Versions of Processors whether there are any performance enhancements we can take advantage of 

**Today:** Few benchmark jobs and Jenkins improvement

### @rtkay123 - Rodney Kanjala

**Yesterday:** Opened PRs for Workflow for the new Repos, linting tested through GitHub workflows, no more lock file issues, Sidecar and  Lumber jack merged

**Today:** Opening PRs today and Passing them for review, session with Shezi for the test suites

### @johanfol - Johan Foley

**Today:** On holiday

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by rtkay123** [comment]
## 2023/12/13

## Attendees
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Reviews PRs from @rtkay123 , completed integration tests in CI for CRSP with `newman`.

**Today:** Use results from newman to post a comment with said content for easier user visibility as for now, you would need to inspect the job from the CI itself. Postponing the merge to main until the team is back

### @vorsterk - Kyle Vorster

**Yesterday:** Working on Jenkins jobs, security related, cleaning up Docker, grype and snyk scans (vulnerabilities on registry and junk cleanup). 

**Today:** Carrying on with security jobs, integrating with Slack for automated feedback. Lookinh into installing the platform through jenkins

### @rtkay123 - Rodney Kanjala

**Yesterday:** Updating actions versions for workflows, codacy kept failing on latest version, artifacts could not be uploaded for scanning, on revert, it works. Tested integration testing CLI for Shezi, connection issues prevented NATS connections. 

**Today:** investigate codacy

### @johanfol - Johan Foley

**Today:** On holiday

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by vorsterk** [comment]
## 2023/12/14

## Attendees
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala

## Apologies
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@johanfol - Johan Foley

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Github workflow , reached the end of the line to competition , adding extras - github artifacts to push benchmarks and compare different performance 

**Today:** Look at artifacts or third party storage will check this today and Kyle to see release.yaml work to help.

### @vorsterk - Kyle Vorster

**Yesterday:** Worked on the security jobs on jenkins , helped Rodney with the github branch actions . Synk scans werent working correctly looking into that and looking into vault and keycloak again for increased security.

**Today:** Working on jobs and vault , keycloak and install nodejs , java and docker on new vm for jenkins.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Working on github actions pipeline  everything's working , dependabot working and triggering through the workflows . Look at vulnerabilities

**Today:** Check the vulnerabilities and dependabot - new library versions.

### @johanfol - Johan Foley

**Today:** On holiday

### @Lenbkr - Len Bekker

**Today:** On holiday

### @JeanPierreNell - Jean-Pierre Nell

**Today:** On holiday

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2024/01/10

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Apologies

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Continued with Node.js CI automated testing - completed CRSP, TADProc and Typology processor. PRs are available for review, showing benchmarks for each of the micro-processors. Updated documentation.

**Today:** Wrap up TMS today.

### @vorsterk - Kyle Vorster

**Yesterday:**

Intended to run some benchmarks, but end-to-end was broken. Implemented the Jenkins process that Len designed. Brandon connected to the container registry. Took only 20 minutes to deploy the platform. Event sidecar is not working at the moment - spent some time with Rodney to try to resolve issues. Rule 90 appeared to be the issue and was fixed - end-to-end working by the end of the day.

**Today:** Benchmark due today, pending fixes to the event side-car.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Updating the rule processor queries for optimization. Debugging the event sidecar.

**Today:** Wrap up rule processor updates.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Yesterday**: Sybrin stuff.

**Today:** 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday**: Sybrin stuff.

**Today:** 

## Parking lot

**Comment by Justus-at-Tazama** [comment]
## 2024/01/11

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Apologies

## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Automated testing of the TMS via the GitHub CI.

**Today:** Wrap up the automated testing work.

### @vorsterk - Kyle Vorster

**Yesterday:** Trouble-shooting the event side-car. The ESC was logging events, but Lumberjack wasn't picking them up. Turned out there was a missing certificate that wasn't recorded into the documentation. Certificate applied. Deployment YAML file updated. Continued benchmarking. Showing a 20/120 EPS increase in performance on the Sybrin hardware. Resource consumption up, but EPS decreased. Paradoxically, an increase in resources results in a decrease in performance.

**Today:** Move benchmarking to Azure.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Trouble-shooting the event side-car and discovered the event.msrc file. Tested the log-level variables, e.g. SILENT works. Attended to some Dependabot merges across processors.

**Today:** Continue rule optimisation.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Yesterday:** KeyCloak investigations. Sybrin stuff.

**Today:**

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Sybrin stuff.

**Today:**

## Parking lot

Please help us pick a name.

**Comment by Justus-at-Tazama** [comment]
## 2023/12/04

## Attendees
@johanfol - Johan Foley
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@rtkay123 - Rodney Kanjala
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell

## Apologies


## Minutes:

### @cshezi - Cebolenkosi Shezi

**Yesterday:** Merging PRs for automated GitHub CI. Some account-cross-over issues when creating PRs from forks.

**Today:** Wrap up today.

### @vorsterk - Kyle Vorster

**Yesterday:** 2200 EPS with 7 fewer VMs and a lower cost of $5000 on 3y reserve. Losing some transactions, but looking into it.

**Today:** Carrying on with Benchmarking. Scaling up some VMs and maxing out processor capacity. Jenkins ran out of storage space - resolving that.

### @rtkay123 - Rodney Kanjala

**Yesterday:** Reviewing rule queries for optimisation opportunities. Implemented some new rule queries. Updated some unit test.

**Today:** Support benchmarking. Update certificate docs for Jumberjack and Elastic.

### @johanfol - Johan Foley

**Yesterday:**

**Today:**

### @Lenbkr - Len Bekker

**Yesterday:** KeyCloak for Sybrin integration

**Today:** 

### @JeanPierreNell - Jean-Pierre Nell

**Yesterday:** Sybrin integration

**Today:** 

## Parking lot

Demo on Monday!
